internal/service/teamprojectassignment/resource.go

```diff
@@ -0,0 +1,177 @@
+package teamprojectassignment
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/validate"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+)
+
+const (
+	resourceName          = "team_project_assignment"
+	errorFetchingResource = "error fetching resource"
+	invalidImportID       = "invalid import ID format"
+	errorAssigment        = "error assigning Team to ProjectID (%s):"
+	errorUpdate           = "error updating TeamID(%s) in ProjectID(%s):"
+	errorDelete           = "error deleting TeamID(%s) from ProjectID(%s):"
+)
+
+var _ resource.ResourceWithConfigure = &rs{}
+var _ resource.ResourceWithImportState = &rs{}
+
+func Resource() resource.Resource {
+	return &rs{
+		RSCommon: config.RSCommon{
+			ResourceName: resourceName,
+		},
+	}
+}
+
+type rs struct {
+	config.RSCommon
+}
+
+func (r *rs) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = resourceSchema()
+	conversion.UpdateSchemaDescription(&resp.Schema)
+}
+
+func (r *rs) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan TFModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connV2 := r.Client.AtlasV2
+	projectID := plan.ProjectId.ValueString()
+	teamID := plan.TeamId.ValueString()
+	teamProjectReq, diags := NewAtlasReq(ctx, &plan)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	_, _, err := connV2.TeamsApi.AddAllTeamsToProject(ctx, projectID, teamProjectReq).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError(fmt.Sprintf(errorAssigment, projectID), err.Error())
+		return
+	}
+
+	apiResp, _, err := connV2.TeamsApi.GetProjectTeam(ctx, projectID, teamID).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError(fmt.Sprintf(errorAssigment, projectID), err.Error())
+		return
+	}
```

Comment:
we can leave a comment that this get would be removed once we have the POST endpoint that adds and returns the single team.